test/cypress/integration/widgets/deployment_actions_spec.js

```diff
@@ -118,25 +130,37 @@ describe('Deployment Action Buttons widget', () => {
                 typeLabelKey('existing_key');
                 typeLabelValue('existing_value');
 
-                cy.get('button.disabled').should('have.attr', 'disabled');
+                cy.get('button[aria-label=Add]').should('have.attr', 'disabled');
             });
             cy.contains('.popup', 'Cannot add the same label twice').should('be.visible');
         });
 
-        it('prevents adding invalid label', () => {
-            function validateIfNotPermitted(string) {
-                typeLabelKey(string);
-                cy.contains('.popup', 'Only letters, digits').should('be.visible');
-            }
-            function validateIfPermitted(string) {
-                typeLabelKey(string);
-                cy.get('.popup').should('not.exist');
+        it('prevents adding label with invalid characters', () => {
+            function checkIfInvalidCharactersPopupIsDisplayed(string) {
+                checkIfPopupIsDisplayed(string, 'Only letters, digits');
             }
+            checkIfPopupIsNotDisplayed('abc-._');
+            checkIfInvalidCharactersPopupIsDisplayed(' ');
+            checkIfInvalidCharactersPopupIsDisplayed('$');
+            checkIfInvalidCharactersPopupIsDisplayed('&');
+        });
 
-            validateIfPermitted('abc-._');
-            validateIfNotPermitted(' ');
-            validateIfNotPermitted('$');
-            validateIfNotPermitted('&');
+        it('prevents adding label with not permitted key', () => {
+            function checkIfInternalKeyIsNotPermitted(key) {
+                checkIfPopupIsDisplayed(key, 'All labels with a `csys-` prefix are reserved for internal use');
+                typeLabelValue('a');
+                cy.get('button[aria-label=Add]').should('have.attr', 'disabled');
+            }
+            function checkIfInternalKeyIsPermitted(key) {
+                checkIfPopupIsNotDisplayed(key);
+                typeLabelValue('a');
+                cy.get('button[aria-label=Add]').should('not.have.attr', 'disabled');
+            }
+            cy.getReservedLabelKeys().then(reservedKeys => {
+                reservedKeys.forEach(key => checkIfInternalKeyIsPermitted(key));
```

Comment:
Optional:

Some code golf :slightly_smiling_face: 

```suggestion
                reservedKeys.forEach(checkIfInternalKeyIsPermitted);
```